tests/conftest.py

```diff
@@ -716,3 +717,65 @@ def rook_repo(request):
     get_rook_repo(
         config.RUN['rook_branch'], config.RUN.get('rook_to_checkout')
     )
+
+
+@pytest.fixture()
+def aws_obj():
+    """
+    Initialize AWS instance
+
+    Returns:
+        AWS: An instance of AWS class
+
+    """
+    aws_obj = aws.AWS()
+    return aws_obj
+
+
+@pytest.fixture()
+def ec2_instances(request, aws_obj):
+    """
+    Get cluster instances
+
+    Returns:
+        dict: The ID keys and the name values of the instances
+
+    """
+    # Get all cluster nodes objects
+    nodes = node.get_node_objs()
+
+    # Get the cluster nodes ec2 instances
+    ec2_instances = aws.get_instances_ids_and_names(nodes)
+    assert ec2_instances, f"Failed to get ec2 instances for node {[n.name for n in nodes]}"
+
+    def finalizer():
+        """
+        Make sure all instances are running
+        """
+        # Getting the instances that are in status 'stopping' (if there are any), to wait for them to
+        # get to status 'stopped' so it will be possible to start them
+        stopping_instances = {
+            key: val for key, val in ec2_instances.items() if (
+                aws_obj.get_instances_status_by_id(key) == constants.INSTANCE_STOPPING
+            )
+        }
+
+        # Waiting fot the instances that are in status 'stopping'
+        # (if there are any) to reach 'stopped'
+        if stopping_instances:
+            for stopping_instance in stopping_instances:
+                instance = aws_obj.get_ec2_instance(stopping_instance.key())
+                instance.wait_until_stopped()
+        stopped_instances = {
+            key: val for key, val in ec2_instances.items() if (
+                aws_obj.get_instances_status_by_id(key) == constants.INSTANCE_STOPPED
+            )
+        }
+
+        # Start the instances
+        if stopped_instances:
+            aws_obj.start_ec2_instances(instances=stopped_instances, wait=True)
+
+    request.addfinalizer(finalizer)
```

Comment:
please move this to the beginning of this fixture

Reply:
But we need the 'nodes' and 'ec2_instances' that are initialized above the finalizer, here.
In addition, everything that takes place above this finalizer doesn't do anything, just init params